File: scripts/verify_langstrecke.js

import { createRequire } from 'module';
const require = createRequire(import.meta.url);
const sqlite3 = require('sqlite3').verbose();
const db = new sqlite3.Database('autox.db');

const query = `
    SELECT 
        r.rank, 
        r.points, 
        c.name as class_name,
        count(*) as count
    FROM race_results r
    JOIN classes c ON r.class_id = c.id
    WHERE LOWER(c.name) LIKE '%langstrecke%' AND r.rank <= 25
    GROUP BY r.rank, r.points, c.name
    ORDER BY r.rank
`;

console.log("Checking Langstrecke Points...");
console.log("Rank | Points | Count | Class");
console.log("-----|--------|-------|------");

db.all(query, [], (err, rows) => {
    if (err) {
        console.error(err.message);
        process.exit(1);
    }

    const rules = {
        1: 40, 2: 35, 3: 30, 4: 27, 5: 25, 6: 23, 7: 21, 8: 19, 9: 17,
        10: 16, 11: 15, 12: 14, 13: 13, 14: 12, 15: 11, 16: 10, 17: 9,
        18: 8, 19: 7, 20: 6, 21: 5, 22: 4, 23: 3, 24: 2, 25: 1
    };

    let fail = false;
    rows.forEach(row => {
        console.log(`${row.rank.toString().padEnd(4)} | ${row.points.toString().padEnd(6)} | ${row.count.toString().padEnd(5)} | ${row.class_name}`);

        const expected = rules[row.rank];
        if (expected && row.points !== expected) {
            console.error(`  MISMATCH: Rank ${row.rank} should have ${expected}`);
            fail = true;
        }
    });

    if (!fail && rows.length > 0) console.log("\nVERIFICATION PASSED for Langstrecke.");
    else if (rows.length === 0) console.log("\nWARNING: No Langstrecke results found to verify.");
    else console.log("\nVERIFICATION FAILED");

    db.close();
});

File: scripts/verify_langstrecke.py
"""Check if Langstrecke driver totals match official PDF"""
import sqlite3

DB_PATH = 'autox.db'
conn = sqlite3.connect(DB_PATH)
cursor = conn.cursor()

print("=== Top Langstrecke Drivers ===")
cursor.execute("""
    SELECT d.name, d.start_number, SUM(rr.championship_points) as total
    FROM race_results rr
    JOIN drivers d ON rr.driver_id = d.id
    WHERE rr.class_id = 'd_lang'
    GROUP BY rr.driver_id
    ORDER BY total DESC
    LIMIT 15
""")
for i, row in enumerate(cursor.fetchall(), 1):
    print(f"  {i}. #{row[1]} {row[0]}: {row[2]} pts")

# Check René Bouma specifically
print("\n=== René Bouma Per-Event ===")
cursor.execute("SELECT id FROM drivers WHERE name LIKE '%Bouma%'")
driver = cursor.fetchone()
if driver:
    cursor.execute("""
        SELECT e.name, rr.rank, rr.points, rr.championship_points, rr.license_type
        FROM race_results rr
        JOIN events e ON rr.event_id = e.id
        WHERE rr.driver_id = ? AND rr.class_id = 'd_lang'
        ORDER BY e.date
    """, (driver[0],))
    total = 0
    for r in cursor.fetchall():
        total += r[3] if r[3] else 0
        print(f"  {r[0]}: rank={r[1]}, pts={r[2]}, champ={r[3]}, license={r[4]}")
    print(f"\nTotal: {total} pts")
    print(f"Expected (from official): 218 pts")

conn.close()
